### ISSUE TITLE
Implement `.__getitem__` for modin's numpy array

### ISSUE THREAD

**Issue by dchigarev** [modin-project/modin#5644]
It would be good if we had at least basic support of the `.__getitem__` operator.
```python
import modin.config as cfg
import modin.pandas as pd

cfg.ExperimentalNumPyAPI.put(True)

df = pd.DataFrame({"a": [1, 2, 3]})
res = df.to_numpy()
print(res[[1, 2]]) # TypeError: 'array' object is not subscriptable
```

**Comment by RehanSD** [comment]
Have a PR for this inflight!

**Comment by noloerino** [comment]
Resolved by #5756; we should file a new issue later if we need more complex indexing.
